src/components/career-form/career-form.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { ErrorMessage, TextArea } from '../form/controls';
+import { Grid, GridItem } from '../grid/grid';
+import axios from 'axios';
+import {
+  Actions,
+  Fieldset,
+  InputField,
+  ProgressBar,
+  SuccessMessage,
+} from './career-components';
+
+interface CareerFormProps {
+  recruiting_channel_id: string;
+  job_position_id: string;
+  access_token: string;
+  company_id: string;
+}
+
+type FormData = {
+  first_name: string;
+  last_name: string;
+  email: string;
+  documents: FileList;
+  message: string;
+  phone?: string;
+};
+
+const API_ENDPOINT = 'https://api.personio.de/recruiting/applicant';
+const SIMPLE_EMAIL_PATTERN = /.+@.+\..+/;
+
+export const CareerForm: React.FC<CareerFormProps> = (props) => {
+  const {
+    register,
+    handleSubmit,
+    setError,
+    clearErrors,
+    errors,
+    formState: { isSubmitSuccessful, isSubmitting },
+  } = useForm();
+
+  const [uploadProgress, setUploadProgress] = useState(0);
+
+  const onSubmit = async (formValues: FormData): Promise<void> => {
+    const apiData = {
+      company_id: props.company_id,
+      access_token: props.access_token,
+      job_position_id: props.job_position_id, // dev test position
+      recruiting_channel_id: props.recruiting_channel_id, //satellytes.com
+    };
+
+    const payload = {
+      ...apiData,
+      ...formValues,
+    };
+
+    const formData = new FormData();
+
+    for (const [key, value] of Object.entries(payload)) {
```

Comment:
I don't really get that for loop, could you add some comments?

Reply:
This is to construct the form data object. I iterate of the `entries` iterator to get my hands on the key & values easily.
Inside the loop I can then differentiate between the `documents` entry (which needs special handling) and the other "atomic" values I can just append as is.